### PR TITLE
👨‍🌾 ui test for adding te with existing client

### LINKED ISSUE #4172: Start TE: Add new project with existing client
Sign in
Start time entry
Enter project creation
Select Existing client
Save Entry
Stop Entry
Successful sync status displayed?

### PR DESCRIPTION
**Pull request by khoi**
## What's this?
ui test for creating te which

- belongs to a project
- the project belongs to an pre-existing client

### Relationships
Closes #4172

## Review hints
None

## :squid: Permissions
🐙

### CODE REVIEW

File: Toggl.Giskard.Tests.UI/AndroidAccessibilityNames.cs [hunk 1]
```diff
@@ -72,11 +72,19 @@ public static class EditTimeEntry
...
+
+    }
+
     public static class NewProject
     {
         public const string ChangeClient = "ChangeClientButton";
         public const string CreateButton = "CreateProjectButton";
         public const string ChangeWorkspace = "ChangeWorkspaceButton";
         public const string TogglePrivateProject = "PrivateProjectSwitchContainer";
+
```
Comment (heytherewill) [inline Toggl.Giskard.Tests.UI/AndroidAccessibilityNames.cs]: Extra newline

File: Toggl.Giskard.Tests.UI/AndroidAccessibilityNames.cs [hunk 2]
```diff
@@ -72,11 +72,19 @@ public static class EditTimeEntry
         public const string DeleteButton = "EditTimeEntryDelete";
     }
 
+    public static class Client
+    {
+        public const string AddFilterTextField = "FilterEditText";
+        public const string ClientCreationCellId = "NameTextView";
+
```
Comment (heytherewill) [inline Toggl.Giskard.Tests.UI/AndroidAccessibilityNames.cs]: Extra newline

File: Toggl.Daneel.Tests.UI/Extensions/StartTimeEntryExtensions.cs [hunk 3]
```diff
@@ -28,5 +28,16 @@ public static void TapCreateClient(this IApp app, string clientName)
         {
             app.TapNthCellInCollection(0);
         }
+
+        public static void TapSelectClient(this IApp app, string clientName)
```
Comment (arturs105) [inline Toggl.Daneel.Tests.UI/Extensions/StartTimeEntryExtensions.cs]: `clientName` isn't used anywhere in the method. This method is supposed to select the correct cell for the client name right?

File: Toggl.Tests.UI/StartTimeEntryTests.cs [hunk 4]
```diff
@@ -113,6 +113,41 @@ public void AddingAClientBeforeSavingPersistsTheClient()
             app.WaitForElement(clientName);
         }
 
+        [Test]
+        public void CreatingATimeEntryWithAnExistingClientWorks()
```
Comment (arturs105) [inline Toggl.Tests.UI/StartTimeEntryTests.cs]: This test doesn't really follow the steps described in the issue - there should be 2 time entries after the test is done.

Comment (khoi) [reply Toggl.Tests.UI/StartTimeEntryTests.cs]: test has been updated to match the described steps

File: Toggl.Daneel.Tests.UI/IosAccessibilityNames.cs [hunk 5]
```diff
@@ -69,11 +69,19 @@ public static class EditTimeEntry
...
+    }
+
     public static class NewProject
     {
         public const string ChangeClient = "ChangeClientButton";
         public const string CreateButton = "CreateProjectButton";
+        public const string CloseButton = "CreateProjectCloseButton";
         public const string ChangeWorkspace = "ChangeWorkspaceButton";
         public const string TogglePrivateProject = "PrivateProjectSwitchContainer";
+        public const string CreateProjectCell = "MvxCreateEntityViewCell";
```
Comment (arturs105) [inline Toggl.Daneel.Tests.UI/IosAccessibilityNames.cs]: I don't like the mcx prefix in this

File: Toggl.Daneel.Tests.UI/Extensions/StartTimeEntryExtensions.cs (hunk 3 above)
Comment (khoi) [reply Toggl.Daneel.Tests.UI/Extensions/StartTimeEntryExtensions.cs]: awesome, thansk for the snippet

**Pull request merged**